cylc/flow/scripts/cylc.py

```diff
@@ -16,10 +16,36 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """cylc main entry point"""
 
-import argparse
-from contextlib import contextmanager
 import os
 import sys
+
+
+def pythonpath_manip():
+    """Stop PYTHONPATH contaminating the Cylc Environment
+
+    * Remove PYTHONPATH items from sys.path to prevent PYTHONPATH
+      contaminating the Cylc Environment.
+    * Re-add items from CYLC_PYTHONPATH to sys.path.
+
+    See Also:
+        https://github.com/cylc/cylc-flow/issues/5124
+    """
+    if 'CYLC_PYTHONPATH' in os.environ:
+        for item in os.environ['CYLC_PYTHONPATH'].split(os.pathsep):
+            abspath = os.path.abspath(item)
+            sys.path.insert(0, abspath)
+    elif 'PYTHONPATH' in os.environ:
```

Comment:
Shouldn't we be removing PYTHONPATH items regardless of whether or not CYLC_PYTHONPATH is defined?

```suggestion
    if 'PYTHONPATH' in os.environ:
```

Reply:
@wxtim - I think this is still wrong, and by the 👍 @oliver-sanders seems to agree. It should be:
- if PYTHONPATH exists, remove its items from sys.path
- if CYLC_PYTHONPATH exists, add its items to sys.path

(not elif).

(And maybe do it in that order too - remove before add - for a tiny efficiency gain)